src/dmd/globals.d

```diff
@@ -183,6 +183,7 @@ extern (C++) struct Param
                             // https://issues.dlang.org/show_bug.cgi?id=16997
     bool fixAliasThis;      // if the current scope has an alias this, check it before searching upper scopes
     bool inclusiveInContracts;   // 'in' contracts of overridden methods must be a superset of parent contract
+    bool noCriticalSectionSize;  // plain 'synchronized { }' statements are run-time allocated
```

Comment:
ca you give this a name that better reflects what it does, perhaps `dynamicallyAllocateCriticalSections`?

Reply:
Any objections in just skipping the preview phase? I imagine nobody is going to use this otherwise (except gdc which will default enable the flag, maybe ldc too if @kinke agrees with this ;-)

Reply:
None here.

Reply:
@thewilsonator - is it worth even bothering with a `-revert`?

Reply:
No but CI is failing
`core.exception.AssertError@src/rt/monitor_.d(213): 0 != true`

Reply:
That was dealt with in latter druntime commit.